pkg/team/api/api.go

```diff
@@ -0,0 +1,484 @@
+		logger:        dopts.logger.Named("api"),
+	}
+}
+
+func (s *TeamService) Register(server *grpc.Server) {
+	proto.RegisterTeamServiceServer(server, s)
+}
+
+func (s *TeamService) CreateTeam(
+	ctx context.Context,
+	req *proto.CreateTeamRequest,
+) (*proto.CreateTeamResponse, error) {
+	localizer := locale.NewLocalizer(ctx)
+	editor, err := s.checkOrganizationRole(
+		ctx, req.OrganizationId,
+		accproto.AccountV2_Role_Organization_MEMBER, localizer)
+	if err != nil {
+		return nil, err
+	}
+	err = s.validateCreateTeamRequest(req, localizer)
+	if err != nil {
+		s.logger.Error(
+			"Failed to validate create team request",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("organizationId", req.OrganizationId),
+				zap.String("name", req.Name),
+			)...,
+		)
+		return nil, err
+	}
+	team, err := domain.NewTeam(
+		req.Name,
+		req.Description,
+		req.OrganizationId,
+	)
+	if err != nil {
+		s.logger.Error(
+			"Failed to create new team",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("organizationId", req.OrganizationId),
+				zap.String("name", req.Name),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	err = s.mysqlClient.RunInTransactionV2(ctx, func(ctxWithTx context.Context, tx mysql.Transaction) error {
+		if err := s.teamStorage.UpsertTeam(ctxWithTx, team); err != nil {
+			return err
+		}
+		// Publish event
+		event, err := domainevent.NewAdminEvent(
+			editor,
+			eventproto.Event_TEAM,
+			team.Id,
+			eventproto.Event_TEAM_CREATED,
+			&eventproto.TeamCreatedEvent{
+				Id:          team.Id,
+				Name:        team.Name,
+				Description: team.Description,
+				CreatedAt:   team.CreatedAt,
+				UpdatedAt:   team.UpdatedAt,
+			},
+			team,
+			team,
+		)
+		if err != nil {
+			return err
+		}
+		return s.publisher.Publish(ctx, event)
+	})
+	if err != nil {
+		s.logger.Error(
+			"Failed to upsert team",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("organizationId", req.OrganizationId),
+				zap.String("name", req.Name),
+			)...,
+		)
+		return nil, s.reportInternalServerError(ctx, err, req.OrganizationId, localizer)
+	}
+	return &proto.CreateTeamResponse{
+		Team: team.Team,
+	}, nil
+}
+
+func (s *TeamService) validateCreateTeamRequest(req *proto.CreateTeamRequest, localizer locale.Localizer) error {
+	if len(strings.TrimSpace(req.Name)) == 0 {
+		dt, err := statusNameRequired.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "name"),
+		})
+		if err != nil {
+			return statusInternal.Err()
+		}
+		return dt.Err()
+	}
+	return nil
+}
+
+func (s *TeamService) DeleteTeam(
```

Comment:
The Members will use the team. So, we can't just delete it without checking it.
How do you plan to implement the teams in the account table?
Are you going to save the team ID, and when listing the accounts, you will join the team table to get the team name?